src/HealthGPS.Console/model_parser.cpp

```diff
@@ -174,43 +177,59 @@ LiteHierarchicalModelDefinition load_dynamic_risk_model_definition(const host::p
+std::shared_ptr<EnergyBalanceModelDefinition>
+load_newebm_risk_model_definition(const host::poco::json &opt) {
     MEASURE_FUNCTION();
-    std::vector<core::Identifier> nutrient_list;
-    std::map<core::Identifier, std::map<core::Identifier, double>> nutrient_equations;
-
-    // Save nutrient identities.
-    auto nutrient_strings = opt["Nutrients"].get<std::vector<std::string>>();
-    for (const std::string &nutrient_str : nutrient_strings) {
-        nutrient_list.emplace_back(nutrient_str);
+    std::unordered_map<core::Identifier, double> energy_equation;
+    std::unordered_map<core::Identifier, std::map<core::Identifier, double>> nutrient_equations;
+    std::unordered_map<core::Gender, std::vector<double>> age_mean_height;
+
+    // Save nutrient -> energy equation.
+    for (const auto &nutrient : opt["Nutrients"]) {
+        auto nutrient_str = nutrient["Name"].get<std::string>();
+        auto nutrient_key = core::Identifier{nutrient_str};
+        auto nutrient_energy = nutrient["Energy"].get<double>();
+        energy_equation[nutrient_key] = nutrient_energy;
     }
 
     // Save food -> nutrient equations.
     for (const auto &food : opt["Foods"]) {
-        auto food_ident = core::Identifier{food["Name"].get<std::string>()};
-        auto food_nutrient_strings = food["Nutrients"].get<std::map<std::string, double>>();
+        auto food_str = food["Name"].get<std::string>();
+        auto food_key = core::Identifier{food_str};
+        auto food_nutrients = food["Nutrients"].get<std::map<std::string, double>>();
 
-        for (const std::string &nutrient_str : nutrient_strings) {
-            auto nutrient_ident = core::Identifier(nutrient_str);
+        for (const auto &nutrient : opt["Nutrients"]) {
+            auto nutrient_str = nutrient["Name"].get<std::string>();
+            auto nutrient_key = core::Identifier{nutrient_str};
 
-            if (food_nutrient_strings.contains(nutrient_str)) {
-                double val = food_nutrient_strings.at(nutrient_str);
-                nutrient_equations[food_ident][nutrient_ident] = val;
+            if (food_nutrients.contains(nutrient_str)) {
+                double val = food_nutrients.at(nutrient_str);
+                nutrient_equations[food_key][nutrient_key] = val;
             }
         }
     }
 
-    return EnergyBalanceModelDefinition(std::move(nutrient_list), std::move(nutrient_equations));
+    // Save M/F average heights for age.
+    auto male_height = opt["AgeMeanHeight"]["Male"].get<std::vector<double>>();
+    age_mean_height.emplace(core::Gender::male, std::move(male_height));
+    auto female_height = opt["AgeMeanHeight"]["Female"].get<std::vector<double>>();
+    age_mean_height.emplace(core::Gender::female, std::move(female_height));
+
+    return std::make_shared<EnergyBalanceModelDefinition>(
+        std::move(energy_equation), std::move(nutrient_equations), std::move(age_mean_height));
```

Comment:
As mentioned, this feels brittle. What if the male and female vectors (why vectors and not arrays, by the way?) have different length? What if they cover just 70 years instead of 100 years? What if they are defined every 2 years instead of every year? Even if this is properly documented, it can break very easily. 

I think there should be a sort of validation step here to make sure the data is consistent, as well as the requirement to include the years in the input file. If they need to be defined every year 0-100, then you can use this step to interpolate/extrapolate the input data.

Reply:
I plan to do a big input sanitisation PR soon, but not against adding the checks now. Why vectors and not arrays? Arrays would indeed be more appropriate here. Force of habit.

Reply:
Ah, it turns out to use `std::array`s, the container size must be be known at compile time (array arguments for functions must be of the form `void fn (std::array<int, ARRAY_SIZE> a) {}`. So we need `std::vector`s here because we don't know upfront how many ages (thus array size) are needed.